LSPS3/README.md

```diff
@@ -0,0 +1,317 @@
+for an `error` field, which exists and is the Boolean value
+`true` if the object is an error object.
+
+Example successful `POST /lsp/jitchannel` response:
+
+```JSON
+{
+    "jit_channel_scid": "1x4815x29451",
+    "lsp_node_id": "03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f",
+    "lsp_cltv_expiry_delta" : 144,
+    "lsp_connection_strings": [
+        "03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f@3.33.236.230:9735"
+    ]
+}
+```
+
+`jit_channel_scid` is the SCID to use for creating the routehint in
+the invoice, `lsp_node_id` is the node for that routehint hop, and
+`lsp_cltv_expiry_delta` is the CLTV delta for that routehint hop.
+
+`lsp_connection_strings` is how the client should connect to the
+LSP.
+
+### 3.  Invoice Generation
+
+The client then generates a [BOLT11][] invoice.
+
+[BOLT11]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/11-payment-encoding.md
+
+* `amount` is the sum of `payment_size` and `opening_fee`.
+  Note that `payment_size` and `opening_fee` are in millisatoshis,
+  but the BOLT11 `amount` uses metric Bitcoin units.
+* The sum of `timestamp` and `x`/`expiry` fields is equal to
+  or less than `opening_fee_valid_until`.
+* `n` is the client node ID.
+* `r` contains a single routehint with a single hop:
+  * `pubkey` is the `lsp_node_id`.
+  * `short_channel_id` is the `jit_channel_scid`.
+  * `fee_base_msat` and `fee_proportional_millionths` are 0.
+  * `cltv_expiry_delta` is `lsp_cltv_expiry_delta`.
+
+All other fields are to be filled in by the client.
+
+The client MAY indicate multipart (`basic_mpp`) support, if it
+supports multipart.
+
+### 4.  Payment
+
+The client MUST connect to the LSP via an indicated
+`lsp_connection_strings`, and then issue the generated invoice
+to the payer of the client.
+
+The payer can then generate one or more outgoing payment parts to
+deliver the total amount to the LSP.
+
+The LSP, when it receives a forward where the next hop has an
+SCID matching an issued `jit_channel_scid`:
+
+* MUST wait for payment parts until all parts sum up to at least
+  `payment_size` plus `opening_fee`.
+* MUST NOT fail with `mpp_timeout`, as the final hop is the client
+  and that error is only allowed at the final hop.
+* MUST fail with `unknown_next_peer` if all parts have not
+  summed up to at least `payment_size` plus `opening_fee` AND
+  `opening_fee_valid_until` has passed.
+* MAY fail with `unknown_next_peer` if it receives too many payment
+  parts.
+* When all conditions below are true, MUST continue to the next
+  step:
+  * The current time is strictly less than `opening_fee_valid_until`.
+  * All received payment parts sum up to at least `payment_size`
+    plus `opening_fee`.
+  * The client is connected.
+
+### 5.  Channel Opening And Forwarding
+
+The LSP requests a channel open to the client.
+The LSP MUST set `option_zeroconf` in the channel opening.
+
```

Comment:
Insert explicitly: The client MAY reject the channel open if the LSP indicates channel parameters in `open_channel` that it disagrees with, which aborts this flow and causes the LSP to reject the payment with `unknown_next_hop`.